content/4_Gold/Modular.mdx

```diff
@@ -367,7 +367,135 @@ We can also use the extended Euclidean algorithm. See the module in the
 
 <CPPOnly>
 
-## Templates
+## With Extended Euclidean Algorithm
+
+The extended Euclidean algorithm computes integers $x$ and $y$ such that
+
+$$
+ax+by=\gcd(a,b)
+$$
+
+
+We can slightly modify the version of the Euclidean algorithm given above to
+return more information!
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+array<ll, 3> extend_euclid(ll a, ll b) {
+	// we know that (1 * a) + (0 * b) = a and (0 * a) + (1 * b) = b
+	array<ll, 3> x = {1, 0, a};
+	array<ll, 3> y = {0, 1, b};
+
+	// run extended Euclidean algo
+	while (y[2] > 0) {
+		// keep subtracting multiple of one equation from the other
+		ll k = x[2] / y[2];
+		for (int i = 0; i < 3; i++) { x[i] -= k * y[i]; }
+		swap(x, y);
+	}
+	return x;  // x[0] * a + x[1] * b = x[2], x[2] = gcd(a, b)
+}
+```
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+static long[] extendEuclid(long a, long b) {
+	// we know that 1 * a + 0 * b = a and 0 * a + 1 * b = b
+	long[] x = {1, 0, a};
+	long[] y = {0, 1, b};
+
+	// run extended Euclidean algo
+	while (y[2] > 0) {
+		// keep subtracting multiple of one equation from the other
+		long k = x[2] / y[2];
+		for (int i = 0; i < 3; i++) { x[i] -= k * y[i]; }
+
+		long[] temp = x;
+		x = y;
+		y = temp;
+	}
+	return x;  // x[0] * a + x[1] * b = x[2], x[2] = gcd(a, b)
+}
+```
+
+</JavaSection>
+
+<PySection>
+
+```py
+def extend_euclid(a: int, b: int) -> list[int]:
+	assert (
+		int(a) > 0 and int(b) > 0
+	), "Arguments must be positive, non-zero numeric values."
+
+	# we know that 1 * a + 0 * b = a and 0 * a + 1 * b = b.
+	x_arr = [1, 0, int(a)]
+	y_arr = [0, 1, int(b)]
+	q = -1
+
+	while y_arr[2] > 0:  # run extended Euclidean algo.
+		q = x_arr[2] // y_arr[2]
+		for i in range(3):
+			# keep subtracting multiple of one equation from the other.
+			x_arr[i] -= y_arr[i] * q
+		x_arr, y_arr = y_arr, x_arr
+
+	return x_arr  # (x[0] * a) + (x[1] * b) = x[2], x[2] = gcd(a, b)
+```
+
+</PySection>
+
+</LanguageSection>
+
+Considering the above code comments and notations, the modular inverse is found in the returned information: $x_0$.
```

Comment:
i mean you could probably just combine these two code snippets into one considering how closely related they are

Reply:
ok moved